kong/pdk/vault.lua

```diff
@@ -597,6 +660,49 @@ local function new(self)
   local _VAULT = {}
 
 
+  local function flush_config_cache(data)
+    local cache = self.core_cache
+    if cache then
+      local vaults = self.db.vaults
+      local old_entity = data.old_entity
+      local old_prefix
+      if old_entity then
+        old_prefix = old_entity.prefix
+        if old_prefix and old_prefix ~= ngx.null then
+          CONFIG_HASHES[old_prefix] = nil
+          cache:invalidate(vaults:cache_key(old_prefix))
+        end
+      end
+
+      local entity = data.entity
+      if entity then
+        local prefix = entity.prefix
+        if prefix and prefix ~= ngx.null and prefix ~= old_prefix then
+          CONFIG_HASHES[prefix] = nil
+          cache:invalidate(vaults:cache_key(prefix))
+        end
+      end
+    end
+
+    LRU:flush_all()
+  end
+
+
+  local function init_worker()
+    _VAULT.init_worker = nil
```

Comment:
This is here so that we can't run this function _twice_. Seems a little drastic to me. At the very least we have to leave a comment for future visitors explaining the possible side-effects of this.

Reply:
I changed it to `initialized` variable.